### ISSUE TITLE
Include HTTP method in cache key

### ISSUE THREAD

**Issue by snarfed** [sh4nks/flask-caching#270]
I hit a bit of a nasty surprise recently when I cached a view like this, and found that `HEAD` responses were being cached and returned to `GET` requests. We should probably include the HTTP method in the cache key by default, right?

```py
@app.route('/foo', methods=['HEAD', 'GET'])
@cache.cached(600)
def foo():
  ...
```

(I know work here is on a bit of a hiatus right now due to #268! Totally ok, just filing to keep track.)

**Comment by nightcityblade** [comment]
Hi, I'd like to work on this. I'll submit a PR shortly.

**Comment by sh4nks** [comment]
This can already be done:

```python
@cache.cached(600, key_prefix=lambda: f"view/{request.method}{request.path}")
def foo(): 
    ...
```

**Comment by snarfed** [comment]
@sh4nks of course! That may be a good enough answer. But fwiw, I'm an experienced developer, and this surprised me. I guess I assumed HTTP method was included in the default cache key. Afaict HTTP method isn't mentioned in the docs anywhere; might be worth adding it, if only to say explicitly that it's not used?

**Comment by sh4nks** [comment]
Hey @snarfed, sorry if my answer did come across a bit harsh - it wasn't meant to be harsh just to answer the question :). Yes good suggestion - I'll add some docs regarding this.
